### ISSUE TITLE
Deprecate / remove DOMUtil

### ISSUE THREAD

**Issue by JoeGermuska** [NUKnightLab/TimelineJS3#799]
The [DomUtil](https://github.com/NUKnightLab/TimelineJS3/blob/master/src/js/dom/DOMUtil.js) class is unnecessary. It has three functions which are better handled using standard DOM Methods:

Calls to [`DOMUtil.addClass(el, name)`](https://github.com/NUKnightLab/TimelineJS3/blob/master/src/js/dom/DOMUtil.js#L12-L16) could be replaced with `el.classList.add(name)`

Calls to [`DOMUtil.removeClass(el,name)`](https://github.com/NUKnightLab/TimelineJS3/blob/master/src/js/dom/DOMUtil.js#L12-L16) could be replaced with `el.classList.remove(name)`

Calls to [`DOMUtil.hasClass(el,name)`](https://github.com/NUKnightLab/TimelineJS3/blob/master/src/js/dom/DOMUtil.js#L7-L10) could be replaced with `el.classList.contains(name)`

Before today, that left only `isInHorizontalViewport(element)`, which is only used in `TimeAxis`... so instead of leaving it hanging around, I just moved it.  

If I felt like we had a better method of testing UX changes, I might have gone ahead and put in those changes above, at least within `DOMUtil` but it feels a bit risky.

This issue's title references deprecation, but I don't even know if that's necessary. Does anyone really use anything that deep in the "public" API for TimelineJS?

(This issue inspired by some code inspection around integrating #798)

**Comment by Dmagical** [comment]
Hey a team of students and I are trying to work on this issue, we are wondering if its just a matter of changing the DOMUtil.js methods with the ones you provided?

**Comment by JoeGermuska** [comment]
@Dmagical yes, I think that would be it. "Deprecate" is probably the wrong word, since we don't have a strong commitment to any public API where we need to ease people into a change.

Ultimately, if the code behaves the same way with things moved out of DOMUtil and that file deleted, then it would be a success. 

PS I am all for students working on open source, so let me know if there are other things we can do to help.

**Comment by meganchung4444** [comment]
Hi! I am on the same team as Dmagical. We are from a program called CTI and had a question! For this issue, we were wondering if we should also change the DOMUtil classes in the website folder as well. Thank you!

**Comment by meganchung4444** [comment]
@JoeGermuska Sorry! I should had been more specific. I made some changes for the function calls in the src/js/timenav and also to the website/static/js/main.js.
I ran npm test and it passed. We just wanted to make sure if we were on the right track and had doubts about changing the website/static/js/main.js code. Thank you for also replying so quick! 

Link to the Page: https://github.com/NUKnightLab/TimelineJS3/blob/master/website/static/js/main.js

**Comment by JoeGermuska** [comment]
@meganchung4444 I wouldn't think that, for this issue, you'd have any reason to change `main.js` but if you'd like to prepare a pull request, we can look at the specifics!

**Comment by Dmagical** [comment]
Hey Joe, I was just wondering, what is the benefit of refactoring the code to use standard DOM Methods instead of the DOMUtil.js methods?

**Comment by JoeGermuska** [comment]
As a general principle, systems with fewer lines of code are easier to understand and reason about when trying to track down bugs. Also, especially when serving users on mobile devices, it's always better to reduce download size to the minimum.

This specific fix is not a big one, but it was an obvious one. There are a lot of other places where TimelineJS probably has more code than needed, both because of coding practices we wouldn't repeat if we started again today, and because since the original code was written, variation between web browsers has become much less, meaning that a lot of code designed to deal with different browsers is redundant now.

Generally, I'd love to prune out a lot of redundant code from TimelineJS, but we don't have very strong testing practices we can use to make sure that after we remove or otherwise refactor code, we haven't potentially "broken" something. Hundreds of thousands of timelines have been created, and we would prefer not to make changes that affect already published timelines as much as possible.

Hope that helps, and happy to answer more questions

**Comment by EdenilsonZetino** [comment]
Hi Mr. Germuska,
I am the other member of the team and I just want to thank you for taking the time to answer our questions and provide guidance along the way.  This is my first open-source project and well it was intimidating however thanks to your approachability and repressiveness you have made it a little less scary.

Thanks again for your time and feedback. I will be looking for more ways to contribute to this project.
